loader.js

```diff
@@ -13,7 +13,14 @@ module.exports = async function mdxEnhancedLoader(src) {
   const options = getOptions(this)
 
   // Parse the front matter
-  const { content, data } = matter(src)
+  let content, data
+  try {
+    const res = matter(src, { safeLoad: true, filename: this.resourcePath })
```

Comment:
Not super important, just curious if the following would work:

```js
const { content, data } = matter(src, { safeLoad: true, filename: this.resourcePath })
```

Reply:
It wouldn't because we are within an async loader and a throw will silently crash - you need to return the error to the callback as is done in the try/catch here

Reply:
It wouldn't because of the scoping - this would create a local copy of `content` and `data` within the try block, since there's `const` before it which indicates initializing a new variable. I tried it without the `const` but it was a syntax error.

Reply:
Ah. Cool, thanks!